### ISSUE TITLE
threadcount vs. go-routines vs. threads vs. go-scheduler

### ISSUE THREAD

**Issue by rcrane** [pingcap/go-ycsb#172]
YCSB allows to define the parameter "threadcount" which, in this (go) code, is used to create go-routines (client.go).

Go-routine don't translate to real threads and are managed by the go-scheduler, as far as I understand.

Is it certain that the parameter 'threadcount' in go-ycsb has the same concurency effect as in the original java-ycsb?
Could the go-scheduler actually interfer and create a concurrency not equal to the concurrency in java-ycsb?

In summary, if I define "-p threadcount=1000" do I really get 1000 concurrent entities that work in parallel or do I get whatever the go-scheduler makes of it?

Does "-p threadcount=1000" imply 1000 connections to the DB?

**Comment by Miro-Puetz** [comment]
Good question! I noticed big performance differences between the java-ycsb and the go-ycsb implementation when I increased the thread count. Maybe thats the reason.
